### PR TITLE
`Assessment`: Fix translation error when deleting manual feedback

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colours only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
The translation when deleting manual feedback during assessment sometimes showed `{{ text }}` instead of the actual feedback text (see screenshot below).

### Description
When using grading instructions, the instruction already provides some text and the feedback detail text is allowed to remain empty. If the text is empty, we now add the grading instruction text to the dialogue.
In case some manual detail text is present, we still use that.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exercise
  - with manual assessment enabled
  - grading instructions (you can add them in the exercise settings)
  - $\geqq 1$ submissions that can be assessed

1. Start the manual assessment.
2. Create a feedback from a grading instruction by dragging the instruction from the right into a feedback (either at the bottom for general feedback, or in case of programming exercises also for inline feedback).
3. Do *not* add additional manual detail text to the feedback.
4. If inline code feedback: Save this feedback.
5. Delete the feedback. You should get a dialogue as shown in the screenshot, but it shows the grading instruction text instead of `{{ text }}`.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
| Class | Coverage | Assertions |
| --- | --- | --- |
| FeedbackContentPipe | 100% | :heavy_check_mark: |
| QuotePipe | 100% | :heavy_check_mark: |

### Screenshots
before:
grafik

@coderabbitai ignore

### CODE REVIEW

**Review by konrad2002 — CHANGES_REQUESTED** [review]
tested on legacy ts11. If you drag and drop the assessment instruction into the feedback area and delete feedback without description, it works as described.

However, I found two issues:

1. when assessing a line in the code, adding feedback without description, and deleting it later, the message still includes `{{ text }}`.
2. when you drag and drop an assessment instruction into an already existing feedback, remove the old description, override the assessment and delete it, no description is shown.
grafik

**Review by konrad2002 — APPROVED** [review]
tested again on legacy ts11, everything works fine now

**Review by LenaKahle — COMMENTED** [review]
Tried testing on ts4 and legacy ts11, both times the drag-and-dropped feedback without additional text can't be saved and when deleting it the prompt is just "Are you sure you want to delete the feedback?" and doesn't contain any text. (When I manually enter text that text is displayed in the dialogue window)
Is there a way to save the feedback without content?

**Comment by b-fein** [comment]
> drag-and-dropped feedback without additional text can't be saved and when deleting it the prompt is just "Are you sure you want to delete the feedback?"

@LenaKahle Sorry about that, I updated the steps for testing. Saving only works for inline code feedback (there is no delete available otherwise). For the general feedback at the bottom the wastebin icon can be used directly without saving.

**Review by chrisknedl — APPROVED** [review]
Code looks good

**Review by LenaKahle — DISMISSED** [review]
Approve, tested on ts4.

**Review by RY997 — DISMISSED** [review]
Code LGTM

**Review by Strohgelaender — DISMISSED** [review]
code

File: src/main/webapp/app/assessment/unreferenced-feedback-detail/unreferenced-feedback-detail.component.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
             <button
                 jhiDeleteButton
                 [buttonSize]="ButtonSize.SMALL"
-                [translateValues]="{ text: feedback.detailText }"
+                [translateValues]="{ text: Feedback.hasContent(feedback) ? ' \'' + Feedback.getContent(feedback) + '\'' : '' }"
```
Comment (rabeatwork) [inline src/main/webapp/app/assessment/unreferenced-feedback-detail/unreferenced-feedback-detail.component.html:13]: Please avoid calling methods from your html template. Instead, you should calculate the desired values beforehand and assign them to a variable.

File: src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-inline-feedback.component.html [hunk 2]
```diff
@@ -66,7 +66,7 @@
                     <button
                         jhiDeleteButton
                         [buttonSize]="ButtonSize.MEDIUM"
-                        [translateValues]="{ text: feedback.detailText }"
+                        [translateValues]="{ text: Feedback.hasContent(feedback) ? ' \'' + Feedback.getContent(feedback) + '\'' : '' }"
```
Comment (rabeatwork) [inline src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-inline-feedback.component.html:69]: Same here

**Review by rabeatwork — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/assessment/unreferenced-feedback-detail/unreferenced-feedback-detail.component.html (hunk 1 above)
Comment (b-fein) [reply src/main/webapp/app/assessment/unreferenced-feedback-detail/unreferenced-feedback-detail.component.html]: Best I can do is put the value in a `get` attribute (see new commit). The user can still change the feedback text and the grading instruction.

**Review by nata-81 — CHANGES_REQUESTED** [review]
Tested on ts2 with a modeling exercise. The text translation works as expected but there is a missing space between the word "feedback" and the grading text. See screenshot: 
image

**Review by nata-81 — APPROVED** [review]
Tested on ts2 works as expected. Except, if there are multiple feedbacks there is a chance that some of them won't show the grading text at all (BUT I DON'T KNOW IF THIS IS RELATED TO THIS PR). See screenshot 1 the first two show no grading text like in picture 2 and the third and the fourth show a grading text like in picture. 
Picture 1: 
image
 picture 2:
image
picture 3:
image

**Review by Gusti2010 — APPROVED** [review]
Tested with a text exercise on t11_Legacy and it works as described. For inline feedback in a test exercise, no delete window is shown at all. However, your test steps specify that we should use inline feedback only for programming exercises, so i think this is fine!
For general feedback it works as expected

**Review by georgyia — APPROVED** [review]
manually tested on ts2, lgtm

**Review by Predixx — APPROVED** [review]
tested and works.
grafik

**Review by Strohgelaender — APPROVED** [review]

**Review by chrisknedl — APPROVED** [review]
Code looks good

**Pull request merged**